UIforETW/Utility.h

```diff
@@ -108,24 +114,24 @@ class ElapsedTimer
 
 // High-precision timer class using QueryPerformanceCounter.
 // This may make ElapsedTimer unnecessary.
-class QPCElapsedTimer
+class QPCElapsedTimer final
 {
 public:
 	QPCElapsedTimer()
 	{
-		QueryPerformanceCounter(&start_);
+		ATLVERIFY(QueryPerformanceCounter(&start_));
 	}
 	double ElapsedSeconds() const
 	{
-		LARGE_INTEGER stop;
-		QueryPerformanceCounter(&stop);
-		LARGE_INTEGER frequency;
-		QueryPerformanceFrequency(&frequency);
+		LARGE_INTEGER stop = {0};

```

Comment:
Actually, the initializers and verifications should just go away. The QueryPerformanceFrequency documentation on MSDN says:
"On systems that run Windows XP or later, the function will always succeed and will thus never return zero."

No point checking for errors that can never happen.